xarray/coding/times.py

```diff
@@ -628,25 +628,38 @@ def _encode_datetime_with_cftime(dates, units: str, calendar: str) -> np.ndarray
     if cftime is None:
         raise ModuleNotFoundError("No module named 'cftime'")
 
+    dates = np.array(dates)
```

Comment:
Not sure what type dates is usually when calling this method. But we could avoid a copy by replacing this by

```suggestion
    dates = np.asarray(dates)
```
For the case that you pass a np.ndarray anyways.